test/test_encryption.py

```diff
@@ -2202,6 +2202,83 @@ def test_05_roundtrip_encrypted_unindexed(self):
         self.assertEqual(decrypted, val)
 
 
+# https://github.com/mongodb/specifications/blob/072601/source/client-side-encryption/tests/README.rst#rewrap
+class TestRewrapWithSeparateClientEncryption(EncryptionIntegrationTest):
+
+    MASTER_KEYS: Mapping[str, Mapping[str, Any]] = {
+        "aws": {
+            "region": "us-east-1",
+            "key": "arn:aws:kms:us-east-1:579766882180:key/89fcc2c4-08b0-4bd9-9f25-e30687b580d0",
+        },
+        "azure": {
+            "keyVaultEndpoint": "key-vault-csfle.vault.azure.net",
+            "keyName": "key-name-csfle",
+        },
+        "gcp": {
+            "projectId": "devprod-drivers",
+            "location": "global",
+            "keyRing": "key-ring-csfle",
+            "keyName": "key-name-csfle",
+        },
+        "kmip": {},
+    }
+
+    def test_rewrap(self):
+        for src_provider in ALL_KMS_PROVIDERS:
+            for dst_provider in ALL_KMS_PROVIDERS:
+                with self.subTest(src_provider=src_provider, dst_provider=dst_provider):
+                    self.run_test(src_provider, dst_provider)
+
+    def run_test(self, src_provider, dst_provider):
+        # Step 1. Drop the collection ``keyvault.datakeys``.
+        self.db.drop_collection("keyvault.datakeys")
```

Comment:
This should drop the datakeys collection in the keyvault database.

Reply:
Fixed